contracts/SpokePool.sol

```diff
@@ -319,14 +330,18 @@ abstract contract SpokePool is
         address originToken,
         uint256 amount,
         uint256 destinationChainId,
-        uint64 relayerFeePct,
-        uint32 quoteTimestamp
+        int64 relayerFeePct,
+        uint32 quoteTimestamp,
+        uint256 maxCount
     ) public payable override nonReentrant unpausedDeposits {
         // Check that deposit route is enabled.
         require(enabledDepositRoutes[originToken][destinationChainId], "Disabled route");
 
         // We limit the relay fees to prevent the user spending all their funds on fees.
-        require(relayerFeePct < 0.5e18, "invalid relayer fee");
+        require(SignedMath.abs(relayerFeePct) < 0.5e18, "Invalid relayer fee");
+        require(amount < 1e36, "Amount too large");
+        require(depositCounter[originToken] <= maxCount, "Above max count");
```

Comment:
Could you not re-use the `depositId` to prevent front-running instead of using the new `depositCounter` per token? i.e. make user pass in `maxDepositId`

Reply:
It's possible, yes. However, I don't think it works as well. Here's the example I've been thinking about:

In normal operation, many users may be wanting to deposit in the same pool at the same time. Instead of setting the deposit count to exactly the current value, they add some cushion to avoid transactions frequently reverting (kind of like uniswap slippage parameters). Let's say my fee is roughly the same as long as there isn't more than 100k of volume that slips in before me. I can then set the current depositCounter + 100k.

Similarly, if I were using depositIds, I could set it to exactly the current value, but if _any_ transaction slips in front of me, my transaction would revert. This would be like setting your uniswap slippage to 0. This would be fine if transactions are well spread out, but if transactions are happening often and someone takes a minute or so to sign their transaction, they will encounter reverts a lot. If I wanted to do something similar to the above for depositIds, I would basically be limiting the _number_ of transactions that come in front of me. Those transactions could be for $1 or $1MM. I have no real way to guarantee that these deposits won't change my quote by more than X.

Does that make sense?

Reply:
hmm I see what you're saying, so `depositCounter` is a much more granular way of protecting against front-running. I just want to balance the gas cost against this added protection as our gas efficiency is a big advantage we have

Reply:
That's fair. I think we should look at gas costs pre and post this change to get a sense of how much it hurts us.